src/plugins/event-plugins/XhrPlugin.ts

```diff
@@ -122,6 +123,14 @@ export class XhrPlugin extends MonkeyPatched<XMLHttpRequest, 'send' | 'open'> {
         ];
     }
 
+    /** Updates endTime if empty and returns it in seconds */
+    private fillEndTimeIfEmpty(xhrDetails: XhrDetails) {
+        if (!xhrDetails.endTime) {
+            xhrDetails.endTime = Date.now();
+        }
+        return xhrDetails.endTime / 1000;
+    }
+
```

Comment:
suggestion: Seems like we've combined a getter and setter method into one method. Can we clarify what singular action this method should do and refactor accordingly? Also want to note that we're setting end time with `ms` as the unit but returning the value in seconds which may be confusing at the first glance. Could we unify to use one unit and do the unit conversion outside this method?

Reply:
* updated to remove `fillEndTimeIfEmpty()`
* moved unit conversion out as well